app/src/main/java/com/github/polybooks/database/FBBookDatabase.kt

```diff
@@ -19,142 +17,127 @@ private const val DATE_FORMAT = "yyyy MM dd"
     private val dateFormatter = SimpleDateFormat(DATE_FORMAT)
 
-    override fun queryBooks(): BookQuery = FBBookQuery()
 
-    private inner class FBBookQuery: AbstractBookQuery() {
-
-        override fun getAll(): CompletableFuture<List<Book>> {
-            when {
-                interests != null -> {
-                    val future = CompletableFuture<List<Book>>()
-                    val hashed = interests!!.map { it.hashCode() }.toList()
-                    bookRef
-                        .whereArrayContainsAny("interests", hashed)
-                        .get().addOnSuccessListener { bookEntries ->
-                            val books = bookEntries.map { bookEntry ->
-                                snapshotEntryToBook(bookEntry)
-                            }
-                            future.complete(books)
-                        }
-                    return future
-                }
-                title != null -> {
-                    val future = CompletableFuture<List<Book>>()
-                    bookRef
-                        .whereGreaterThanOrEqualTo("book.title", title!!)
-                        .whereLessThanOrEqualTo("book.title", title!! + '\uf88f')
-                        .get().addOnSuccessListener { bookEntries ->
-                            val books = bookEntries.map { bookEntry ->
-                                snapshotEntryToBook(bookEntry)
-                            }
-                            future.complete(books)
-                        }
-                    return future
-                }
-                isbns != null -> {
-                    val isbns = this.isbns!!
-                    val booksFromFBFuture = getBooksByISBNFromFirebase(isbns.toList())
-                    return booksFromFBFuture.thenCompose { booksFromFB ->
-                        val isbnsFound = booksFromFB.map { it.isbn }
-                        val remainingISBNs = isbns.minus(isbnsFound)
-                        val booksFromOLFuture = bookProvider.queryBooks().searchByISBN(remainingISBNs).getAll()
-                        val allBooksFuture = booksFromOLFuture.thenApply { booksFromOL ->
-                            booksFromOL + booksFromFB
-                        }
-                        val booksToFBFuture = booksFromOLFuture.thenCompose { booksFromOL ->
-                            val futures = booksFromOL.map { book -> addBookToFirebase(book) }
-                            listOfFuture2FutureOfList(futures)
-                        }
-                        booksToFBFuture.thenCompose { allBooksFuture }
-                    }
+    override fun searchByTitle(
+        title: String,
+        ordering: BookOrdering
+    ): CompletableFuture<List<Book>> {
+        val future = CompletableFuture<List<Book>>()
+        bookRef
+            .whereGreaterThanOrEqualTo("book.title", title)
+            .whereLessThanOrEqualTo("book.title", title + '\uf88f')
```

Comment:
if you search for "boo", it will return books with titles between "boo" and "booz" in the dictionnary. the '\uf88f' is just a very large character. The goal is to accept any book whose title starts with "boo".